### PR TITLE
kibana: provisioning

### PR DESCRIPTION
**Pull request by alissa-tung**

### CODE REVIEW

File: pkg/service/elk_service.go [hunk 1]
```diff
@@ -127,6 +129,12 @@ func (k *Kibana) Deploy(globalCtx *GlobalCtx) *executor.ExecuteCtx {
 	}
 	args := spec.GetDockerExecCmd(globalCtx.containerCfg, k.spec.ContainerCfg, k.ContainerName, true, mountPoints...)
 	args = append(args, k.spec.Image)
+
+	if k.spec.ProvisioningTemplate != "" {
+		args = append(args, "&&")
+		args = append(args, "/usr/bin/env -S bash", filepath.Join(k.CheckReadyScriptPath))
```
Comment (YangKian) [inline pkg/service/elk_service.go:135]: you have a check-ready function, why still need this?

File: pkg/service/elk_service.go [hunk 2]
```diff
@@ -150,12 +158,51 @@ func (k *Kibana) SyncConfig(globalCtx *GlobalCtx) *executor.TransferCtx {
 	position := []executor.Position{
 		{LocalDir: genCfg, RemoteDir: filepath.Join(k.spec.RemoteCfgPath, "kibana.yml")},
 	}
+	if k.spec.ProvisioningTemplate != "" {
+		position = append(position, executor.Position{LocalDir: fmt.Sprintf("template/kibana/%s.ndjson", k.spec.ProvisioningTemplate),
```
Comment (YangKian) [inline pkg/service/elk_service.go:162]: spec.ProvisionTemplate should already contain the `.ndjson` suffix

File: pkg/service/elk_service.go [hunk 3]
```diff
@@ -150,12 +158,51 @@ func (k *Kibana) SyncConfig(globalCtx *GlobalCtx) *executor.TransferCtx {
...
+			RemoteDir: filepath.Join(k.spec.RemoteCfgPath, fmt.Sprintf("%s.ndjson", k.spec.ProvisioningTemplate))})
+	}
+
+	if k.spec.ProvisioningTemplate != "" {
```
Comment (YangKian) [inline pkg/service/elk_service.go:166]: remove duplicated `if` command

File: pkg/service/elk_service.go [hunk 4]
```diff
@@ -150,12 +158,51 @@ func (k *Kibana) SyncConfig(globalCtx *GlobalCtx) *executor.TransferCtx {
...
+	}
 
 	return &executor.TransferCtx{
 		Target: k.spec.Host, Position: position,
 	}
 }
 
+func (k *Kibana) CheckReady(globalCtx *GlobalCtx) *executor.ExecuteCtx {
+	if len(k.CheckReadyScriptPath) == 0 {
+		panic("empty checkReadyScriptPath")
```
Comment (YangKian) [inline pkg/service/elk_service.go:198]: return nil

**Review by YangKian — CHANGES_REQUESTED** [review]

File: pkg/service/elk_service.go (hunk 1 above)
Comment (alissa-tung) [reply pkg/service/elk_service.go]: thx [0abc458](https://github.com/hstreamdb/deployment-tool/pull/28/commits/0abc4580c438c0e15de46119e42d3ee275582383)

File: pkg/service/elk_service.go (hunk 2 above)
Comment (alissa-tung) [reply pkg/service/elk_service.go]: thx ef514cb

File: pkg/service/elk_service.go (hunk 3 above)
Comment (alissa-tung) [reply pkg/service/elk_service.go]: thx ef514cb

**Comment by alissa-tung** [comment]
r+ @YangKian  tests passed

**Review by YangKian — APPROVED** [review]

**Pull request merged**